django/db/models/query_utils.py

```diff
@@ -169,6 +173,78 @@ def _check_parent_chain(self, instance, name):
         return None
 
 
+class RegisterLookupMixin(object):
+    def _find_lookup(self, lookup_name, class_attr):
+        """
+        Finds lookups or transforms registered on this, or any parent, class.
+        `class_attr` can be either `class_lookups` or `class_transforms`.
+        """
+        try:
+            return getattr(self, class_attr)[lookup_name]
+        except KeyError:
+            # To allow for inheritance, check parent class'
+            for parent in inspect.getmro(self.__class__):
+                parent_lookups = getattr(parent, class_attr, None)
+                if not parent_lookups:
+                    continue
+                if lookup_name in parent_lookups:
+                    return parent_lookups[lookup_name]
+        except AttributeError:
+            # Nothing has been registered
+            pass
+        return None
+
+    def get_lookup(self, lookup_name):
+        found = self._find_lookup(lookup_name, 'class_lookups')
+        if found is None and hasattr(self, 'output_field'):
+            return self.output_field.get_lookup(lookup_name)
+        return found
+
+    def get_transform(self, lookup_name):
+        found = self._find_lookup(lookup_name, 'class_transforms')
+        if found is None and hasattr(self, 'output_field'):
+            return self.output_field.get_transform(lookup_name)
+        return found
+
+    @classmethod
+    def register_lookup(cls, lookup, lookup_name=None):
+        from django.db.models.lookups import Transform
+        if issubclass(lookup, Transform):
+            # TODO (Josh): should we deprecate this method of registering transforms?
+            # Backwards compatibility -- allows original register_lookup to
+            # register transforms of type `Transform`. New code should use
+            # `register_transform`.
+            return cls.register_transform(lookup, lookup_name)
+        if lookup_name is None:
+            lookup_name = lookup.lookup_name
+        if 'class_lookups' not in cls.__dict__:
+            cls.class_lookups = {}
+        cls.class_lookups[lookup_name] = lookup
+        return lookup
+
+    @classmethod
+    def register_transform(cls, transform, lookup_name=None):
+        if lookup_name is None:
+            lookup_name = transform.lookup_name
+        if 'class_transforms' not in cls.__dict__:
```

Comment:
Not in this case, no. A getattr/hasattr will check subclasses too which is not what we want here.